### ISSUE TITLE
[SC2126] Shouldn't be a warning when using `-r` and `-R` with grep

### ISSUE THREAD

**Issue by andlrc** [koalaman/shellcheck#803]
#### For bugs
- Rule Id (if any, e.g. SC2126): 
- My shellcheck version 0.4.5
- I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
grep -r pattern . | wc -l
grep -R pattern . | wc -l

```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
 
Line 2:
grep -R a . | wc -l
^-- SC2126: Consider using grep -c instead of grep|wc.
 
Line 3:
grep -r a . | wc -l
^-- SC2126: Consider using grep -c instead of grep|wc.
```

#### Here's what I wanted or expected to see:

{NOTHING}

Using `-r` or `-R` (follow symbolic links) shouldn't yield this warning because the output from (GNU) grep is something like:

    $ grep -r pattern directory
    file1: line with pattern
    file1: pattern in line
    file2: line containing pattern
    file3: pattern contained in line
    $ grep -rc pattern directory
    file1: 2
    file2: 1
    file3: 1

While the output from piping to `wc -l` would be:

    $ grep -r pattern directory | wc -l
    4

I tried with `--no-filename`:

    $ grep --no-filename -rc pattern directory
    2
    1
    1
